### PR TITLE
Handle dead job status outside of transactions

### PR DESCRIPTION
**Pull request by kalikiana**
As per https://metacpan.org/pod/DBIx::Class::Storage#txn_do results assuming a transaction was executed should only be logged after the entire coderef was executed.

See: https://progress.opensuse.org/issues/121768

### CODE REVIEW

File: lib/OpenQA/Scheduler/Model/Jobs.pm [hunk 1]
```diff
@@ -468,24 +468,21 @@ sub incomplete_and_duplicate_stale_jobs ($self) {
...
-                    );
-                    my $res = $job->auto_duplicate;
-                    if (ref $res) {
-                        log_warning(sprintf('Dead job %d aborted and duplicated %d', $job->id, $res->id));
-                    }
-                    else {
-                        log_warning(sprintf('Dead job %d aborted as incomplete', $job->id));
-                    }
-                });
+            log_warning(
```
Comment (kalikiana) [inline lib/OpenQA/Scheduler/Model/Jobs.pm:471]: Obvious drawback here is that all 3 cases share the same log level. But the message only comes up after the transaction finished successfully if I understand the docs correctly.

**Review by Martchus — CHANGES_REQUESTED** [review]
This only goes one level but log messages from e.g. `auto_duplicate` are still unaffected. Printing some log messages only after the transaction and some during the transaction makes things just more inconsistent in my opinion. Printing log messages only after the transaction makes it also hard to follow what was happening within the transaction.
